java-checks-test-sources/src/main/java/checks/BigDecimalDoubleConstructorCheck.java

```diff
@@ -13,14 +13,28 @@
   BigDecimal bd2 = new BigDecimal(2.0); // Noncompliant [[sc=20;ec=39;quickfixes=qf1]] {{Use "BigDecimal.valueOf" instead.}}
   // fix@qf1 {{Replace with BigDecimal.valueOf}}
   // edit@qf1 [[sc=20;ec=34]] {{BigDecimal.valueOf}}
-  BigDecimal bd4 = new BigDecimal(2.0, mc); // Noncompliant [[sc=20;ec=43;quickfixes=!]] {{Use "BigDecimal.valueOf" instead.}}
+  BigDecimal bd4 = new BigDecimal(2.0, mc); // Noncompliant [[sc=20;ec=43;quickfixes=qf1_2]] {{Use "BigDecimal.valueOf" instead.}}
+  // fix@qf1_2 {{Replace with BigDecimal("2.0")}}
```

Comment:
The replacement message is misleading, it shows a missing `MathContext` argument.
And without the MathContext, the value is not the same, for example:
```
    MathContext mathContext = new MathContext(3, RoundingMode.FLOOR);
    System.out.println(new BigDecimal(1.2345f, mathContext)); // => 1.23
    System.out.println(new BigDecimal("1.2345")); // => 1.2345
```
We could perhaps say `Replace with BigDecimal("2.0",`